VogtObserver3/Store.cs

```diff
@@ -9,22 +9,34 @@ class Store : IObserveable
 
         private readonly List<Observer> _observers = new List<Observer>();
 
+        private float _discount;
+
         public void AddObserver(Observer o)
         {
             _observers.Add(o);
         }
 
-        public void Notify()
+        public void Notify(float discount)
         {
             foreach (var observer in _observers)
             {
-
+                observer.Update(discount);
             }
         }
 
         public void RemoveObserver(Observer o)
         {
             _observers.Remove(o);
         }
+
+
+        public void ApplyDiscount()
+        {
+            Random random = new Random();
+
+            _discount = random.Next(1, 55);
```

Comment:
Based on my last comment, I think you can simplify this with:

Notify(random.Next(1, 55));